### PR TITLE
=BG= extract build task

### PR DESCRIPTION
**Pull request by bguiz**
- experimental, now in separate node module

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -39,13 +40,9 @@ if (configTaskPlugins.constructor !== Array) {
...
 }
 
-var pluginContext = {
-  toolPath: __dirname
-};
-
 pluginRegistry
   .get('angularity')
-  .context(pluginContext)
+  .context({})
```
Comment (bguiz) [inline index.js]: `toolPath` is no longer required due to improements in `plugin-registry`

File: index.js [hunk 2]
```diff
@@ -61,9 +58,23 @@ pluginRegistry
...
+            return taskYargsRun;
+          default:
+            throw new Error('Context dependency ' + name + ' is unknown');
+        }
+      },
       taskYargsRun: taskYargsRun,
-      gulp: gulp,
-      runSequence: runSequence
+      gulp: require('gulp'),
+      runSequence: require('run-sequence')
```
Comment (bguiz) [inline index.js]: all these dependencies need not be defined here, and only within `getDependency`, **after** all tasks have been converted

**Comment by bholloway** [comment]
This raises some big issues:
* I am committed to splitting out tasks however I'm fundamentally opposed to loosing their GIT history.
* There are issues surrounding incremental build (that @impaler has highlighted) we need to solve before we can say that splitting out tasks will work. 
* I think the modules need some wider review and it is easier to do this before we split them.
* I believe that CI coverage is the higher priority. I would prefer to 'swarm' these issues (as @debabrata would say) so that we can complete quicker.

I do not want to be obstructive but we need to evaluate these issues.

**Comment by bholloway** [comment]
I think we need to close, if only until we resolve the above.

**Pull request closed without merging**